src/StackEmptyException.java

```diff
@@ -0,0 +1,12 @@
+public class StackEmptyException extends Throwable {
```

Comment:
You subclassed Throwable here rather than Exception. Seems like it could mean more work in real world examples.